Mathlib/Data/Finset/LocallyFinite/Box.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2024 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.Ring.Prod
+import Mathlib.Data.Int.Interval
+import Mathlib.Order.Disjointed
+import Mathlib.Tactic.Ring
+import Mathlib.Tactic.Zify
+
+/-!
+# Decomposing a locally finite ordered ring into boxes
+
+This file proves that any locally finite ordered ring can be decomposed into "boxes", namely
+differences of consecutive intervals.
+
+## Implementation notes
+
+We don't need the full ring structure, only that there is an embedding `ℕ → `
```

Comment:
Does this work at least for `OrderedSemiring`? Or maybe even for additive monoid with 1 (of characteristic zero?).

Reply:
No, you also need a negation. Let me clarify